### DISCUSSION TITLE
How to integrate the quality factor (similar to pagerank) into the ranking of Qdrant search results?

### DISCUSSION THREAD

**Discussion by wacdev** [qdrant/qdrant#1801] [Q&A]
For example, the sorting result I expect is based on (0.5*correlation+0.5*quality factor), is there a way for me to do this simply when querying the Qdrant database? Or should I take all the search results and calculate the order myself?

My requirement is just a simple text-image search, I don't want to introduce too many components.

**Comment by generall** [comment]
We do not recommend to use something like `0.5correlation+0.5quality factor`  as a ranking factor and consider it as a bad practice.
Qdrant does not support this type of scoring out of the box, but you can still do a re-scoring on the client side, if you are sure that it is something you need

**Reply by wacdev** [reply]
If it is 0.5correlation+0.5quality factor is a bad practice.

So how to introduce the quality factor? For example, if I do image search, some images are beautiful and some are ugly. How can this be reflected in the ranking of search results?

**Reply by generall** [reply]
There are several options:

- Incorporate "beautifulness" into the query. Some naive approach could be to simply add an extra keyword to the query.
- Use filters, e.g. everything below 80% is ugly and stored down. You can make a batch search query and apply multiple conditions
- Actually re-rank results with a model, which knows how to combine `correlation` and `quality`. 


Any solution you will find out there, that promises to re-rank by coefficients will simply apply those scores on a post-processing stage. It will be effectively the same as just re-score the top-N results on the client side.
